### ISSUE TITLE
Code not working

### ISSUE THREAD

**Issue by umesh-21** [ShimingWang98/Facial_Feature_Crafting_against_Inversion_based_Identity_Theft#2]
Please update the latest and running version of code. As I earlier also mention that your readme file not properly guide exactly which encoder parameter need to download if we perform backbox attack. Also check whitebox code like train_gan.py there generator initiaiztion is completely wrong because you not define or importing any generator module. Kindly update latest and working version of code

**Comment by ShimingWang98** [comment]
- **Encoder parameters.**  
    The _Inference Scenario_ of the README already states that:

    > Here you should have a trained encoder and a downstream classifier, with models defined in Crafter_protection/models.py and parameters stored under params/. If not, try  
    > ```bash
    > $ cd Crafter_protection/
    > $ python trainTargetNet.py -whole
    > ```
    > It will train a split ResNet18 as a demo.

    By default, the encoder parameters are saved to `params/enc.pt` (see [trainTargetNet.py](https://github.com/ShimingWang98/Facial_Feature_Crafting_against_Inversion_based_Identity_Theft/blob/main/Crafter_protection/trainTargetNet.py), lines 62–65), which matches the load path in the black-box attack code ([trainDecoder.py](https://github.com/ShimingWang98/Facial_Feature_Crafting_against_Inversion_based_Identity_Theft/blob/main/BlackBoxAttack/trainDecoder.py), line 45).

2. **Generator initialization.**
In train_gan.py, you have`from models import *`, which imports the `GanGenerator()` module from [models.py](https://github.com/ShimingWang98/Facial_Feature_Crafting_against_Inversion_based_Identity_Theft/blob/main/models.py) and initializes the generator.

**Comment by umesh-21** [comment]
class GanGenerator(nn.Module): #in_dim=100
    def __init__(self, in_dim=100, dim=64):
        super(GanGenerator, self).__init__()

        def dconv_bn_relu(in_dim, out_dim):
            return nn.Sequential(
                nn.ConvTranspose2d(in_dim, out_dim, 5, 2,
                                   padding=2, output_padding=1, bias=False),
                nn.BatchNorm2d(out_dim),
                nn.ReLU())

        self.l1 = nn.Sequential(
            nn.Linear(in_dim, dim * 8 * 4 * 4, bias=False),
            nn.BatchNorm1d(dim * 8 * 4 * 4),
            nn.ReLU())
        self.l2_5 = nn.Sequential(
            dconv_bn_relu(dim * 8, dim * 4),
            dconv_bn_relu(dim * 4, dim * 2),
            dconv_bn_relu(dim * 2, dim),
            nn.ConvTranspose2d(dim, 3, 5, 2, padding=2, output_padding=1),
            nn.Sigmoid())

    def forward(self, x):
        y = self.l1(x)
        y = y.view(y.size(0), -1, 4, 4)
        y = self.l2_5(y)
        return y

above given inside models.py

but as used in train_gan.py utilized different parameters here given example below

generator = GanGenerator(z_dim=latent_dim, y_dim=2) in this z_dim , and y_dim not define 

and during calling 

 fake_imgs = generator(z, cond)

here its used 2 parameter but in forward process its required only one parameter

**Comment by ShimingWang98** [comment]
Code mismatch fixed in commit [74c63c0](https://github.com/ShimingWang98/Facial_Feature_Crafting_against_Inversion_based_Identity_Theft/commit/74c63c04809810a21e597cbae7c38104c01ac733). Thanks for noticing. Also feel free to check out the original GAN inversion repo [here](https://github.com/AI-secure/GMI-Attack) for the white-box attacker's snippet.
